### DISCUSSION TITLE
Maintaining "Focus"

### DISCUSSION THREAD

**Discussion by Bumblebee001** [GrandOrgue/GoOdf#63] [General]
Hi..... 

1. When working with stops one often is dealing with the "Internal Rank" tab features. When a different stop is selected, the focus falls back to the first tab (Stop Properties). I think it would be more useful to maintain the focus on the last chosen tab irrespective of which stops are selected, until that focus is changed by user.

2. Also the behaviour of the cursor selection should be corrected when using the arrow keys to go up and down the various elements in the lists within the left "window".

3. Another useful feature would be to be able to drag the cursor (left click to select) over multiple elements on the panel layout screen to move them 'en bloc' eg when one wants to move several stops together. Doing them one by one is not only much more time consuming but a confusion of overlapping elements will result as they are being re-arranged.

**Comment by larspalo** [comment]
1. I think that making the stop panel remember which of the notebook pages was last active wouldn't be too much of a hassle, but not all stops utilize internal ranks. If a new stop is selected, that use the opposite (internal rank or referenced rank) of the last, should one then revert to the first stop notebook page as is currently done, or should one jump to the equivalent used page?

2. I'm not sure if you're repeating the same concerns as I answered in https://github.com/larspalo/GOODF/discussions/49#discussioncomment-6689609 with this. Some of the elements are simply not possible to navigate to as they are only used for grouping sub items. That's why there will be jumps. I'm sure that the present (default) keyboard navigation of the organ tree could be improved upon, but I've really not had the time to think about it.

3. I've not forgotten that suggestion. It's just that there are so many other things to do and allowing multi element selection on the rendered panel window is quite a bit more complicated than handling only a single element.

**Reply by Bumblebee001** [reply]
It's OK Lars.... I just mention things I think could be improved from the noob-user-perspective. ;-)

**Reply by larspalo** [reply]
The number one point is included in the 0.8.0 release and I've added the number three point as planned for the future.

**Comment by Bumblebee001** [comment]
Thanks
